esx_service/utils/auth_data.py

```diff
@@ -734,24 +734,29 @@ def err_config_init_needed(self):
         return error_code_to_message[ErrorCode.INIT_NEEDED]
 
 
-    def new_db(self):
+    def __init_db(self):
```

Comment:
ok, this is what I had in mind. If __connect() fails, there is nothing to delete. 
So 
```python

def __init_db(self):  # this is literally cut-n-paste of old code in new_db()
  """Inits freshly created DB by creating needed tables."""
  err = self.__create_tables()
  if err:
     return err

   err = self.__create_default_tenant()
   if err:
      return err

    err = self.__create_all_ds_privileges_for_default_tenant()
    if err:
      return err

    err = self.__create_vm_ds_privileges_for_default_tenant()
    if err:
       return err


def new_db(self): 
        """
        Create brand new DB content at self.db_path. Expects clean slate.
        :returns: None for success , a string (with error) for error
        """
        if not self.conn:
                # old DB will be opened on a new DB file will be created. 
                # We do not expect the file for already-exists, but if it does , __init_db below will fail.
                # Note that if something is wrong, this will throw so no err checking
                self.__connect()          

       err = __init_db()
       if err:
            os.remove(self.db_path)
            return err
```